substrate/frame/ranked-collective/src/lib.rs

```diff
@@ -840,6 +848,91 @@ pub mod pallet {
 		}
 	}
 
+	#[cfg(any(feature = "try-runtime", test))]
+	impl<T: Config<I>, I: 'static> Pallet<T, I> {
+		/// Ensure the correctness of the state of this pallet.
+		pub fn do_try_state() -> Result<(), sp_runtime::TryRuntimeError> {
+			Self::try_state_members()?;
+			Self::try_state_index()?;
+
+			Ok(())
+		}
+
+		/// ### Invariants of Member storage items
+		///
+		/// Total number of [`Members`] in storage should be >= [`MemberIndex`] of a [`Rank`] in
+		///    [`MemberCount`].
+		/// [`Rank`] in Members should be in [`MemberCount`]
+		fn try_state_members() -> Result<(), sp_runtime::TryRuntimeError> {
+			MemberCount::<T, I>::iter().try_for_each(|(_, member_index)| -> DispatchResult {
+				let total_members = Members::<T, I>::iter().count();
+				ensure!(
+				total_members as u32 >= member_index,
+				"Total count of `Members` should be greater than or equal to the number of `MemberIndex` of a particular `Rank` in `MemberCount`."
+				);
+
+				Ok(())
+			})?;
+
+			Members::<T, I>::iter().try_for_each(|(_, member_record)| -> DispatchResult {
+				ensure!(
+					Self::is_rank_in_member_count(member_record.rank.clone().into()) == true,
+					"`Rank` in Members should be in `MemberCount`"
+				);
+
+				Ok(())
+			})?;
+
+			Ok(())
+		}
+
+		/// ### Invariants of Index storage items
+		/// [`Member`] in storage of [`IdToIndex`] should be the same as [`Member`] in [`IndexToId`]
+		/// [`Rank`] in [`IdToIndex`] should be in [`IndexToId`]
+		fn try_state_index() -> Result<(), sp_runtime::TryRuntimeError> {
+			IdToIndex::<T, I>::iter().try_for_each(
+				|(rank, who, member_index)| -> DispatchResult {
+					let who_from_index = IndexToId::<T, I>::get(rank, member_index).unwrap();
+					ensure!(
+				who == who_from_index,
+				"`Member` in storage of `IdToIndex` should be the same as `Member` in `IndexToId`."
+				);
+
+					ensure!(
+						Self::is_rank_in_index_to_id_storage(rank.into()) == true,
+						"`Rank` in `IdToIndex` should be in `IndexToId`"
```

Comment:
This doesn't make sense to me, maybe rephrase and/or add a comment

Reply:
so, for this @liamaharon https://github.com/paritytech/polkadot-sdk/pull/3007#discussion_r1479291547

The sum of all member_indexes will not equal total_members because, the indexes for a particular rank will keep increasing and not decrease. for example look at this test https://github.com/paritytech/polkadot-sdk/blob/master/substrate/frame/ranked-collective/src/tests.rs#L456 the total_members are 3 and the sum of all the member_indexes is 9 because for every add or promote that is being done, therefore, the index of each rank will keep increasing if the member is in a particular rank or have been promoted. So, if a member was added at rank 0 and the promoted to rank 1, the index of rank 0 is 1 and the index of rank 1 is also 1 too, making the total index 2 for that particular member

Reply:
I have now rephrased this.